src/caches.jl

```diff
@@ -153,8 +179,16 @@ function OrdinaryDiffEq.alg_cache(
     C_2DxD = zeros(uElType, 2D, D)
     C_3DxD = zeros(uElType, 3D, D)
 
-    backward_kernel = AffineNormalKernel(
-        zeros(uElType, D, D), zeros(uElType, D), PSDMatrix(zeros(uElType, 2D, D)))
+    backward_kernel = if KRONECKER
+        AffineNormalKernel(
+            kronecker(_I(d), zeros(uElType, q+1, q+1)),
+            zeros(uElType, D),
+            PSDMatrix(kronecker(_I(d), zeros(uElType, 2*(q+1), q+1)))
+        )
+    else
+        AffineNormalKernel(
+            zeros(uElType, D, D), zeros(uElType, D), PSDMatrix(zeros(uElType, 2D, D)))
+    end
 
```

Comment:
At this point I'm wondering: Should there be a `initialize_covariance_matrix(::CovarianceFactorization, dims...)`? This would scale to a future block-diagonal!